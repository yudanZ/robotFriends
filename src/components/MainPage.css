.container{
    margin-top: 10px;
}
h1{
    font-family: 'SEGA LOGO FONT';
    color:rgb(55, 140, 236);
    font-weight: 200;
    font-size: 2rem;
}

@font-face {
    font-display: swap;
    font-family: 'SEGA LOGO FONT';
    font-style: normal;
    font-weight: normal;
    src: local('SEGA LOGO FONT'), url('../containers/SEGA.woff') format('woff');
    }