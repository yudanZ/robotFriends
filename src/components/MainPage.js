import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundry';
import './MainPage.css';


class MainPage extends Component {
   constructor(props){
       super(props)
   }
    componentDidMount(){
        //console.log(this.props.store.getState());
        this.props.onRequestRobots();
        
        //console.log('check');
        //this.setState({robots: robots});
    }
    filteredRobots = () => {
        return this.props.robots.filter( robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
        })
    }
    
    render (){
      
        const {  onSearchChange, isPending} = this.props;
        
      
        if( isPending ){
            return <h1>Loading</h1>
        }else {
            return (
                <div className="container">
                    <h1 className="text-center">RoboFriends</h1>
                    <SearchBox searchChange={onSearchChange}/>
                    <Scroll>
                        <ErrorBoundary>
                            <CardList robots={this.filteredRobots()}/>
                        </ErrorBoundary>
                        
                    </Scroll>
                    
                </div>
               
            
            );
        }
        //console.log(filteredRobots);
        

    }
    
}

export default MainPage;