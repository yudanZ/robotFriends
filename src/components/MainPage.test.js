import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots : jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage { ...mockProps} />)
})
it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filters Robots correctly' , () => {
    const mockProps2 = {
        onRequestRobots : jest.fn(),
        robots: [{
            id:3,
            name:'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage { ...mockProps2} />)

    expect(wrapper2.instance().filteredRobots()).toEqual([{
        id:3,
        name:'John',
        email: 'john@gmail.com'
    }])
})

it('filters Robots empty',() => {
    const mockProps3 = {
        onRequestRobots : jest.fn(),
        robots: [{
            id:3,
            name:'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    }
    const wrapper3 = shallow(<MainPage { ...mockProps3} />)
    expect(wrapper3.instance().filteredRobots()).toEqual([])
})

it('isPending is true',() => {
    const mockProps4 = {
        onRequestRobots : jest.fn(),
        robots: [{
            id:3,
            name:'John',
            email: 'john@gmail.com'
        }],
        searchField: '',
        isPending: true
    }
    const wrapper4 = shallow(<MainPage { ...mockProps4} />)
    expect(wrapper4.find('h1')).toHaveLength(1);
})