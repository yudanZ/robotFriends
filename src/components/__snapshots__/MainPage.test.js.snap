// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
<div
  className="container"
>
  <h1
    className="text-center"
  >
    RoboFriends
  </h1>
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
